test/rules/src/https_everywhere_checker/rules.py

```diff
@@ -179,6 +181,28 @@ def _determineTestApplication(self):
 				self.determine_test_application_run = True
 		return self.test_application_problems
 
+	def getTargetValidityProblems(self):
+		"""Verify that each target has a valid TLD in order to prevent problematic rewrite
+			 as stated in EFForg/https-everywhere/issues/10877. In particular, 
+			 right-wildcard target are ignored from this test.
+
+			 Returns an array of strings reporting any coverage problems if they exist,
+			 or empty list if coverage is sufficient.
+			 """
+		problems = self._determineTestApplication()
+		# Next, make sure each target has a valid TLD
+		for target in self.targets:
+			# Ignore right-wildcard targets
+			if target.endswith(".*"):
+				continue
```

Comment:
Since IPv4 and IPv6 addresses are [valid subjects](https://tools.ietf.org/html/rfc3280), we should add a conditional here which ignores these targets as well.